tests/download/download_test.py

```diff
@@ -0,0 +1,248 @@
+    assert os.path.exists(os.path.join(final_dir, f"{title}.zip"))
+
+
+def test_download_feature_failure(mocker: Any, tmp_path: Path) -> None:
+    title = "S2B_MSIL1C_20241209T162609_N0511_R040_T17UPV_20241209T195414.SAFE"
+    mock_feature = {
+        "id": "a6215824-704b-46d7-a2ec-efea4e468668",
+        "properties": {
+            "title": title,
+            "collection": "SENTINEL-2",
+            "services": {"download": {"url": "http://example.com"}},
+        },
+    }
+    mocker.patch(
+        "cdsetool.download.download_file",
+        side_effect=lambda url, path, options: None,
+    )
+
+    final_dir = str(tmp_path / "test_download_feature_failure")
+    filename = download_feature(mock_feature, final_dir)
+    assert filename is None
+
+
+def test_download_feature_with_filter(mocker: Any, tmp_path: Path) -> None:
+    options = {"filter_pattern": "*.jp2"}
+    title = "S2B_MSIL1C_20241209T162609_N0511_R040_T17UPV_20241209T195414.SAFE"
+    mock_feature = {
+        "id": "a6215824-704b-46d7-a2ec-efea4e468668",
+        "properties": {
+            "title": title,
+            "collection": "SENTINEL-2",
+        },
+    }
+    mocker.patch(
+        "cdsetool.download.filter_files",
+        return_value=["./GRANULE/file1.jp2", "./GRANULE/file2.jp2"],
+    )
+    mocker.patch(
+        "cdsetool.download.download_file",
+        mock_download_file,
+    )
+
+    final_dir = str(tmp_path / "test_download_feature_with_filter")
+    product_name = download_feature(mock_feature, final_dir, options)
+    assert os.path.exists(os.path.join(final_dir, title, "GRANULE", "file1.jp2"))
+    assert os.path.exists(os.path.join(final_dir, title, "GRANULE", "file2.jp2"))
+    assert product_name == mock_feature["properties"]["title"]
+
+
+def test_download_feature_with_filter_failure(mocker: Any, tmp_path: Path) -> None:
+    options = {"filter_pattern": "*.jp2"}
+    mock_feature = {
+        "id": "a6215824-704b-46d7-a2ec-efea4e468668",
+        "properties": {
+            "title": "S2B_MSIL1C_20241209T162609_N0511_R040_T17UPV_20241209T195414.SAFE",
+            "collection": "SENTINEL-2",
+        },
+    }
+    mocker.patch(
+        "cdsetool.download.filter_files",
+        return_value=["./GRANULE/file1.jp2", "./GRANULE/file2.jp2"],
+    )
+    mocker.patch(
+        "cdsetool.download.download_file",
+        side_effect=lambda url, path, options: None,
+    )
+
+    final_dir = str(tmp_path / "test_download_feature_with_filter_failure")
+    product_name = download_feature(mock_feature, final_dir, options)
+    assert product_name is None
+
+
+def test_download_feature_with_filter_unsupported_coll(
+    caplog: Any, tmp_path: Path
+) -> None:
+    options = {"logger": logging.getLogger(__name__), "filter_pattern": "*MTL.txt"}
+    mock_feature = {
+        "id": "a6215824-704b-46d7-a2ec-efea4e468668",
+        "properties": {
+            "title": "L8XXX",
+            "collection": "Lansat8",
```

Comment:
Missing `d` in the collection name, and try to remove the trailing comma and see if this formats better.